File: hw/jacome/U3/HW22Strategy/JS/view/SortApp.js
const SortingContext = require('../controller/SortingContext');
const readline = require('readline');

function printArray(array) {
    console.log(array.join(' '));
}

function main() {
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });
    
    const sortingContext = new SortingContext();
    
    rl.question('Ingrese el numero de elementos: ', (n) => {
        const numElements = parseInt(n);
        const data = [];
        let count = 0;
        
        console.log(`Ingrese ${numElements} numeros (uno por línea):`);
        
        rl.on('line', (input) => {
            data.push(parseInt(input));
            count++;
            
            if (count === numElements) {
                console.log('Array original:');
                printArray(data);
                
                const sortedData = sortingContext.sort([...data]);
                
                console.log('Array ordenado:');
                printArray(sortedData);
                
                rl.close();
            }
        });
    });
    
    rl.on('close', () => {
        process.exit(0);
    });
}

main();

File: hw/astudillo/U3/HW21Strategy/Strategy/src/main/java/ec/edu/espe/template/strategy/view/SortApp.java
package ec.edu.espe.template.strategy.view;

import ec.edu.espe.template.strategy.controller.SortingContext;
import java.util.Scanner;

/**
 *
 * @author PC
 */
public class SortApp {
    
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        SortingContext sortingContext = new SortingContext();
        
        System.out.print("Ingrese el numero de elementos: ");
        int n = scanner.nextInt();
        
        int[] data = new int[n];
        System.out.println("Ingrese " + n + " numeros:");
        
        for (int i = 0; i < n; i++) {
            data[i] = scanner.nextInt();
        }
        
        System.out.println("Array original:");
        printArray(data);
        
        int[] sortedData = sortingContext.sort(data.clone());
        
        System.out.println("Array ordenado:");
        printArray(sortedData);
        
        scanner.close();
    }
    
    private static void printArray(int[] array) {
        for (int i = 0; i < array.length; i++) {
            System.out.print(array[i] + " ");
        }
        System.out.println();
    }
}
